daemon/src/monitor.rs

```diff
@@ -554,77 +554,77 @@ impl xtra::Actor for Actor {
         self.tasks
             .add(this.clone().send_interval(Duration::from_secs(20), || Sync));
 
-        self.tasks.add_fallible(
-            {
-                let db = self.db.clone();
-                let this = this.clone();
+        self.tasks.add({
+            let db = self.db.clone();
+            let this = this.clone();
 
-                async move {
-                    let mut stream = db.load_all_open_cfds::<Cfd>(());
+            async move {
+                let mut stream = db.load_all_open_cfds::<Cfd>(());
 
-                    while let Some(Cfd {
+                while let Some(cfd) = stream.next().await {
+                    let Cfd {
                         cet,
                         commit_tx,
                         lock_tx,
                         ..
-                    }) = stream
-                        .try_next()
-                        .await
-                        .context("Failed to load CFD from database")?
-                    {
-                        if let Some(tx) = commit_tx {
-                            if let Err(e) = this
-                                .send(TryBroadcastTransaction {
-                                    tx,
-                                    kind: TransactionKind::Commit,
-                                })
-                                .await?
-                            {
-                                tracing::warn!("{e:#}")
-                            }
+                    } = match cfd {
+                        Ok(cfd) => cfd,
+                        Err(e) => {
+                            tracing::warn!("Failed to load CFD from database: {e:#}");
+                            continue;
                         }
-
-                        if let Some(tx) = cet {
-                            if let Err(e) = this
-                                .send(TryBroadcastTransaction {
-                                    tx,
-                                    kind: TransactionKind::Cet,
-                                })
-                                .await?
-                            {
-                                tracing::warn!("{e:#}")
-                            }
+                    };
+
+                    if let Some(tx) = commit_tx {
+                        if let Err(e) = this
+                            .send(TryBroadcastTransaction {
+                                tx,
+                                kind: TransactionKind::Commit,
+                            })
+                            .await
+                            .expect("receiver to outlive the actor")
```

Comment:
❓ I didn't expect to see this change in this commit. Can we undo it or at least move it to a separate commit?

Reply:
After having another look I am actually not sure we *should* expect in these cases. 

These are cases where we send to ourselves from a task that we spawn. I think we can assume that the actor should always be alive in such a case (because the task is cleaned up with the actor).
But probably it's still better to treat this as it was. I removed the expects - which means that in `monitor` the tasks added remain fallible.